src/routes/add-invoice.test.tsx

```diff
@@ -0,0 +1,121 @@
+import { renderWithRouter, screen } from './../test/test-utils.tsx';
+import { describe, expect, it } from 'vitest';
+import { within } from '@testing-library/react';
+import { addDays, format } from 'date-fns';
+import App from '../App.tsx';
+
+describe('AddInvoicePage', () => {
+  const { user } = renderWithRouter(<App />);
```

Comment:
If you are going to render whole App, then there is no need to use `renderWithRouter`, you can use simple `render`. RenderWithRouter wraps it with additional Providers, which duplicates all contexts (Router, QueryClient etc)